### ISSUE TITLE
Intersection summary and TCS reports failing at Bloor and High Park 

### ISSUE THREAD

**Issue by candu** [CityofToronto/bdit_flashcrow#715]
**What Happened**
On [View Reports at Bloor and High Park](https://move.intra.dev-toronto.ca/view/location/s1:A4WvmB/POINTS/reports/TMC), the TMC Summary and TMC Detailed reports work, but the Intersection Summary and warrant reports do not.

**What Should Happen**
All four report types should work at all intersections.

**To Reproduce**
Steps to reproduce the bug:
1. Go to [View Reports at Bloor and High Park](https://move.intra.dev-toronto.ca/view/location/s1:A4WvmB/POINTS/reports/TMC)
2. Click on the "Intersection Summary Report" tab near top.

At this point, the loading spinner remains indefinitely, and the report does not load.

**Screenshots**
Loading spinner:

![loading spinner for Intersection Summary report at Bloor and High Park](https://user-images.githubusercontent.com/151605/100618228-8b411880-32e9-11eb-982a-3c9b950bad49.png)

Stacktrace from `reporter` for intersection summary report:

```
Debug: handler, error
    TypeError: Cannot read property 'short' of null
    at Function.computeHourlyTotals (/var/app/flashcrow/lib/reports/ReportIntersectionSummary.js:80:56)
    at /var/app/flashcrow/lib/reports/ReportIntersectionSummary.js:162:50
    at Array.map (<anonymous>)
    at ReportIntersectionSummary.transformData (/var/app/flashcrow/lib/reports/ReportIntersectionSummary.js:161:37)
    at ReportIntersectionSummary.generate (/var/app/flashcrow/lib/reports/ReportBase.js:126:34)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async handler (/var/app/flashcrow/lib/controller/ReportController.js:121:26)
    at async exports.Manager.execute (/var/app/flashcrow/node_modules/@hapi/hapi/lib/toolkit.js:60:28)
    at async Object.internals.handler (/var/app/flashcrow/node_modules/@hapi/hapi/lib/handler.js:46:20)
    at async exports.execute (/var/app/flashcrow/node_modules/@hapi/hapi/lib/handler.js:31:20)
```

The warrant report stacktrace is similar, with the same `TypeError` in `computeHourlyTotals`.

**Additional Notes**
Before the weekend, at the same location, these were loading successfully.  There appear to be new counts at the same location, but this error is also observed in the old counts.  Perhaps something changed in the centreline?

Digging deeper into the code, the offending line is:

```js
const majorLeftMax = data[`${majorLeftMaxDirection.short}_L`];
```

`majorLeftMaxDirection` itself is computed using:

```js
const majorLeftMaxDirection = ArrayUtils.getMaxBy(
  majorDirections,
  ({ short: majorDir }) => data[`${majorDir}_L`],
);
```

which only returns `null` if `majorDirections.length === 0`.  `majorDirections` itself is computed by `ReportBaseFlowDirectional.getDirectionalStats` on a per-hour basis, using `roadId` to group legs of an intersection.  `roadId` itself is an alias for the `lfn_id` column in `gis.centreline`.

**Comment by candu** [comment]
The intersection in question has [`centrelineId 13466332`](https://move.intra.dev-toronto.ca/api/locations/byCentreline?s1=s1:A4WvmB).  It has these adjoining legs:

```sql
flashcrow=> select geo_id, lfn_id from gis.centreline where fnode = 13466332 or tnode = 13466332 order by lfn_id asc;
  geo_id  | lfn_id
----------+--------
 10250061 |    458
 30119699 |    458
  7730504 |   3197
  7994458 |   3660
(4 rows)
```

The list in our dev dataset, which was last loaded into my dev environment a few weeks back, is slightly different:

```
flashcrow=> select geo_id, lfn_id from gis.centreline where fnode = 13466332 or tnode = 13466332 order by lfn_id asc;
  geo_id  | lfn_id 
----------+--------
  1145070 |    458
 10250061 |    458
  7730504 |   3197
  7994458 |   3660
(4 rows)
```

Here the leg with `geo_id = 30119699` has `geo_id = 1145070` instead - this feature does not appear in the newer dataset.  It's not yet clear if (or how) this affects report generation...

**Comment by candu** [comment]
This same issue occurs at:

- [Don Mills and Overlea](https://move.intra.dev-toronto.ca/view/location/s1:A8gqmB/POINTS/reports/TMC)
- [Overlea and Thorncliffe](https://move.intra.dev-toronto.ca/view/location/s1:AsuqmB/POINTS/reports/TMC)
- [Danforth and Logan](https://move.intra.dev-toronto.ca/view/location/s1:A4PtmB/POINTS/reports/TMC)

It does not occur at:

- [Pape and Danforth](https://move.intra.dev-toronto.ca/view/location/s1:AeKtmB/POINTS/reports/TMC)
- [Howard and Sherbourne](https://move.intra.dev-toronto.ca/view/location/s1:AC2tmB/POINTS/reports/TMC)
- [Bloor and Church](https://move.intra.dev-toronto.ca/view/location/s1:Aq3tmB/POINTS/reports/TMC)

In each case, it appears that the issue either occurs for all counts at that location, or does not occur for all counts at that location.  This adds further support to the hypothesis that this is somehow related to centreline geometry, and not to the count data itself.

**Comment by candu** [comment]
The count at Don Mills and Overlea from 2015-05-20 is present in both AWS dev and local dev.  However, the issue shows up in AWS dev, and not in local dev!  The count data appears identical in both environments.

This *may* be a floating-point issue: `GeometryUtils.getLineStringBearingFrom` expects `lineString` to have an endpoint that _exactly_ matches `point`.  However, looking at [Don Mills and Overlea](https://move.intra.dev-toronto.ca/api/locations/byCentreline?s1=s1:A8gqmB) vs. [the Overlea segment](https://move.intra.dev-toronto.ca/api/locations/byCentreline?s1=s1:AdWBSB), the segment has higher precision.

**Comment by candu** [comment]
After the fix in #716 , this issue is no longer showing up on AWS dev - marking closed.
